packages/runtime/container-runtime/src/dataStores.ts

```diff
@@ -205,14 +211,59 @@ export class DataStores implements IDisposable {
                 }),
             pkg);
 
-        // Resolve pending gets and store off any new ones
         this.contexts.addBoundOrRemoted(remotedFluidDataStoreContext);
 
         // Equivalent of nextTick() - Prefetch once all current ops have completed
         // eslint-disable-next-line @typescript-eslint/no-floating-promises
         Promise.resolve().then(async () => remotedFluidDataStoreContext.realize());
     }
 
+    public processAliasMessage(message: ISequencedDocumentMessage): IDataStoreAliasMapping | undefined {
+        const aliasMessage = message.contents as IDataStoreAliasMessage;
+        if (!isDataStoreAliasMessage(aliasMessage)) {
+            throw new DataCorruptionError(
+                "malformedDataStoreAliasMessage",
+                {
+                    ...extractSafePropertiesFromMessage(message),
+                },
+            );
+        }
+
+        const existingMapping = this.aliasMap.get(aliasMessage.alias);
+        if (existingMapping !== undefined) {
+            return {
+                suppliedInternalId: aliasMessage.internalId,
+                alias: aliasMessage.alias,
+                aliasedInternalId: existingMapping,
+            };
+        }
+
+        // Unlikely scenario, but we may have an alias OP with the alias value
+        // equal to the id of an older named data store
+        const maybeContextWithAliasAsId = this.contexts.get(aliasMessage.alias);
```

Comment:
Do you mean - user intentionally used internal ID of data store as an alias?
I have two thoughts on this:
- long term, it should be impossible to even get into this situation, as we should not expose internal IDs (only handles). It would be great to have a comment somewhere that early return means failure.
- Short term, that operation should fail. It's not clear from this code whether it's the case. I assume so (as we do not reach the end of this function).

Reply:
This scenario is when a user creates a named root data store using our existing APIs. This would have the `internald` (the id field of the data store context object) be a custom string supplied by the user. If we ship aliasing and the user tries to alias a datastore to the same name which was used to create a previous root data store, we need to catch it here in order to avoid a conflict. 

For example:

`createRootDataStore(pkg, "myDataStore"` would create it and it would be stored in the document. That Id should be forever excluded from aliasing so that if the user now does `createRootDataStore().Alias("myDataStore")` , the call should fail.

Reply:
Maybe then update a comment above? I'm sure people will wonder looking at the code and not sure if it "unlikely scenario".